streaming/src/main/scala/org/apache/gearpump/streaming/dsl/task/GroupByTask.scala

```diff
@@ -61,13 +62,14 @@ class GroupByTask[IN, GROUP, OUT](
   }
 
   override def onWatermarkProgress(watermark: Instant): Unit = {
-    groups.values.forEach(new Consumer[WindowRunner[IN, OUT]] {
-      override def accept(runner: WindowRunner[IN, OUT]): Unit = {
-        runner.trigger(watermark).foreach {
-          result =>
-            taskContext.output(Message(result.value, result.timestamp))
+    if (groups.isEmpty && watermark == Watermark.MAX) {
```

Comment:
Other the `processingWatermark` will never advance when there are no incoming messages and `groups.isEmpty`